src/avram/errors.cr

```diff
@@ -59,4 +59,22 @@ module Avram
       super message
     end
   end
+
+  class ConnectionError < AvramError
+    def initialize(connection_details : URI)
+      error = String.build do |message|
+        message << "Failed to connect to databse '#{connection_details.path.try(&.[1..-1])}' with username #{connection_details.user}.\n"
+        message << "You can try these steps to help:"
+        message << '\n'
+        message << '\n'
```

Comment:
I think these should be double quotes but I am not sure.

Reply:
They should be single because the carriage return is a single character. It's actually more performant to do it this way though negligible when talking about this small of code...

Reply:
TIL!